addons/decimal_precision/models/decimal_precision.py

```diff
@@ -13,6 +13,25 @@ class DecimalPrecision(models.Model):
         ('name_uniq', 'unique (name)', """Only one value can be defined for each given usage!"""),
     ]
 
+    @api.onchange('digits')
+    def _onchange_digits(self):
+        new_rounding = 1.0 / 10.0**self.digits
+        dangerous_uom = self.env['product.uom'].search([('rounding', '<', new_rounding)])
+        if dangerous_uom:
+            errors = ["'%s' (id=%s, precision=%s)." % (uom.name, str(uom.id), str(uom.rounding)) for uom in dangerous_uom]
+            warning = {
+                'title': _('Warning!'),
+                'message':
+                    _(
+                        "You are setting a Decimal Accuracy less precise than"
+                        " the UOM:\n %s \n"
+                        "This may cause inconsistencies in reservations.\n"
+                        "Please set a higher number of Digits."
```

Comment:
Since it's truncate to decimal precision and not UoM rounding this indication is probably not enough.
Something like:
Please set a higher number of Digits. -> Please increase the rounding of this unit of measure and the global decimal precision.